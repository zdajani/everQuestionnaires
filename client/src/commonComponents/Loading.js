import React from 'react';
import './styles/Loading.css';

const Loading = () => (
  <div className="loading-screen">
  	<img src={'https://tools.cisco.com/security/center/Resources/images/loader.gif'} alt="loadig" id="loading"/>
  </div>
);

export default Loading;