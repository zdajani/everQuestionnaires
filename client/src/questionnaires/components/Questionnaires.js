import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom'
import QuestionnairesList from '../components/QuestionnairesList'
import './styles/Questionnaires.css'; 

const Questionnaires = ({ questionnaires, url }) => {
  const isAdminPage = url.includes('admin');
  return (
    <div>
      <section className="jumbotron text-center">
        <div className="container">
          <h1 className="jumbotron-heading page-title">{isAdminPage ? "Admin Page" : "everQuestionnaires"}</h1>
          <p className="lead text-muted">
            Here for your questionnaire needs anytime anywhere
          </p>
          <div className="top-buttons">
            <Link to='/questionnaires/new' className="btn btn-outline-info">
              <h4>Create A Questionnaire</h4>
            </Link>
          </div>
      { !isAdminPage &&
          <div className="top-buttons">
            <Link to='/admin/questionnaires' className="btn btn-outline-info">
              <h4>View Answers To Your Questionnaires</h4>
            </Link>
          </div> }
        </div>
      </section>
        {questionnaires && <QuestionnairesList questionnaires={questionnaires} url={url} isAdminPage={isAdminPage} />}
    </div>
  );
}

Questionnaires.propTypes = {
  questionnaires: PropTypes.objectOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired
  }).isRequired),
  url: PropTypes.string
};

export default Questionnaires;