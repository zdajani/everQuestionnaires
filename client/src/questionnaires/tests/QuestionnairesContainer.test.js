import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import ConnectedQuestionnairesContainer, { QuestionnairesContainer } from '../containers/QuestionnairesContainer';
import Questionnaires from '../components/Questionnaires';
import Loading from '../../commonComponents/Loading';

const middlewares = []
const mockStore = configureMockStore(middlewares);

const questionnairesData = {
  1: { id: 1, 
  title: "Lord of the Rings",
  created_at: "2017-08-15T18:24:26.586Z",
  updated_at: "2017-08-15T18:24:26.586Z" }
}

// setup to return QuestionnairesContainer not the one wrapped by connected()
// this so to test just the rendering of the component
const setup = (questionnairesData, isLoading) => {
  const match = {url: '/questionnaires'}
  const container = shallow(
      <QuestionnairesContainer 
        questionnaires={questionnairesData} 
        isLoading={isLoading}
        match={match}
        />
  );
    
  return {
    container,
    match
  };
}

// setup to return wrapped QuestionnairesContainer
// this is used to test its interaction with Redux
const connectedSetup = (store) => {
  const connectedContainer = shallow(
      <ConnectedQuestionnairesContainer store={store}/>
  );
  return {
    connectedContainer
  };
}

describe('Questionnaires container', () => {
  it('renders without crashing', () => {  
    const store = mockStore({
      questionnaires: {
        isLoading: false, 
        data: null,
        errorMessage: null
      },
    });
    
    const fetchQuestionnairesSpy = jest.fn();
    const match = {url: '/questionnaires'}
    // using mount for Full DOM rendering to make sure 
    //the component doesn't crash 
    const mountedComponent = mount(<QuestionnairesContainer store={store} fetchQuestionnaires={fetchQuestionnairesSpy} match={match} isLoading={true}/>);
    
    expect(toJson(mountedComponent)).toMatchSnapshot();
  });
  
  describe('when questionnaires data is available', () => {
    const store = mockStore({
      questionnaires: {
        isLoading: false, 
        data: questionnairesData,
        errorMessage: null
      },
    });
    
    it('renders self with questionnaires props', () => {    
      const { connectedContainer } = connectedSetup(store);  
      expect(connectedContainer).toHaveProp('questionnaires', questionnairesData);
      expect(toJson(connectedContainer)).toMatchSnapshot();
    });
    
    it('renders Questionnaires component', () => {   
      const isloading = false; 
      const { container } = setup(questionnairesData, isloading);  
      const questionnaires = container.find(Questionnaires)
      expect(toJson(questionnaires)).toMatchSnapshot();
    });
    
    it('does not render Loading component', () => {    
      const { container } = setup(questionnairesData);       
  
      expect(container).not.toContainReact(<Loading />);
    });
  });
  
  describe('when loading', () => {
    const data = null;
    const isloading = true;     
    const { container } = setup(data, isloading); 
    
    it('renders loading component', () => {  
      expect(container).toContainReact(<Loading />);
    });
    
    it('does not render Questionnaires component', () => {  
      expect(container).not.toContainReact(
        <Questionnaires questionnaires={questionnairesData} url='/questionnaires'/>
      );
    });
  });
});