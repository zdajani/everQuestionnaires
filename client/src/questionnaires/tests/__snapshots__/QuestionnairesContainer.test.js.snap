// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questionnaires container renders without crashing 1`] = `
<QuestionnairesContainer
  fetchQuestionnaires={[Function]}
  isLoading={true}
  match={
    Object {
      "url": "/questionnaires",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div>
    <Loading>
      <div
        className="loading-screen"
      >
        <img
          alt="loadig"
          id="loading"
          src="https://tools.cisco.com/security/center/Resources/images/loader.gif"
        />
      </div>
    </Loading>
  </div>
</QuestionnairesContainer>
`;

exports[`Questionnaires container when questionnaires data is available renders Questionnaires component 1`] = `
<Questionnaires
  questionnaires={
    Object {
      "1": Object {
        "created_at": "2017-08-15T18:24:26.586Z",
        "id": 1,
        "title": "Lord of the Rings",
        "updated_at": "2017-08-15T18:24:26.586Z",
      },
    }
  }
  url="/questionnaires"
/>
`;

exports[`Questionnaires container when questionnaires data is available renders self with questionnaires props 1`] = `
<QuestionnairesContainer
  fetchQuestionnaires={[Function]}
  isLoading={false}
  questionnaires={
    Object {
      "1": Object {
        "created_at": "2017-08-15T18:24:26.586Z",
        "id": 1,
        "title": "Lord of the Rings",
        "updated_at": "2017-08-15T18:24:26.586Z",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
